src/lib/core/NodeId.h

```diff
@@ -89,4 +97,25 @@ constexpr uint16_t PAKEKeyIdFromNodeId(NodeId aNodeId)
     return aNodeId & kMaskPAKEKeyId;
 }
 
+constexpr bool IsValidCaseNodeId(NodeId aNodeId)
```

Comment:
This seems like a pretty ACL-specific concept of "valid CASE node id"...  Really, it's IsValidCASESubject, and it should probably not be in this header.

Reply:
It does seem though like there are some parts of the range of 64-bits which are invalid, and hence there should be some "IsValid" functions that belong here, even if it's not specifically this one. I do think we should define some.

Reply:
I think "IsValidNodeId" is meaningful in general, yes....

Reply:
I moved them into Access Control since they are probably more applicable there. When we want more general ones, we can add them more generally.